common/flatpak-dir-private.h

```diff
@@ -936,24 +944,44 @@ typedef struct
   char *ref;
   char *opt_commit;
 
+  /* Used during p2p resolve */
+  OstreeCollectionRef collection_ref; /* owns the collection_id member only, ref_name is from above */
+  char               *local_commit;
+  char               *latest_remote_commit;
+
   /* out */
   char   *resolved_commit;
   GBytes *resolved_metadata;
   guint64 download_size;
   guint64 installed_size;
   char   *eol;
   char   *eol_rebase;
+  gint32  token_type;
```

Comment:
Maybe I missed discussion of this somewhere else in this PR, but I was hoping to understand the semantics of this a bit more. From [the flat-manager side](https://github.com/flatpak/flat-manager/pull/29#issuecomment-552645947):

> I’ll have a more detailed look later, but for the token type there are two answers. First the technical, token type zero is “behave as before” (ie no tokens needed), and anything else is undefined in general. Any non zero value triggers an authenticator call passing the ref and the token type. It is then up to the Authenticator what to do about it. It could even return an empty token if it want to signal that no token was needed anyway.
>
> The second answer is how I imagine flathub to use this. I think we’ll have two values. One for “needs purchase” and one for “needs donation flow”. Both of these will just run some webflow that end up in a possibly empty token. However the purchase one will start with a login and will always try to get a token, whereas the donation one will uses an anonymous webflow the first time and then track locally that it already did so so it can be skipped the next time.

Perhaps this should be a flags value rather than a specific value? Seems like that might be a little more flexible down the road.

Reply:
Not sure what you mean by "flags value" here. Its intentionally an opaque (except zero) integer to flatpak in order to allow each authenticator to decide how to use this as best fits the requirements for that particular backend.